packages/flutter_tools/lib/src/asset.dart

```diff
@@ -434,8 +446,7 @@ class ManifestAssetBundle implements AssetBundle {
     for (final _Asset asset in materialAssets) {
       final File assetFile = asset.lookupAssetFile(_fileSystem);
       assert(assetFile.existsSync(), 'Missing ${assetFile.path}');
-      entries[asset.entryUri.path] ??= DevFSFileContent(assetFile);
-      entryKinds[asset.entryUri.path] ??= asset.assetKind;
+      entries[asset.entryUri.path] ??= AssetBundleEntry(DevFSFileContent(assetFile), kind: asset.kind);
```

Comment:
This is an example of code where it would have been easy to forget to provide the optional `kind` argument.